Clarified and added detail for devices, customizations and groups

lindsaymarkward opened the pull request:
added example of group views that refer to other groups.

I've been going through the docs and using the Gitter chat to understand them, so I'm updating the docs to be clearer. 
(I think more examples of configuration files with images to match would be *really* helpful.)

balloob commented on source/_components/group.markdown:
````diff
@@ -52,14 +53,24 @@ Configuration variables:
 
 ```yaml
-# Example configuration.yaml entry
-group:
````
These configuration examples are here to show that there are three different ways to specify a group config.

balloob commented on source/getting-started/devices.markdown:
```diff
@@ -36,21 +36,22 @@ switch:
 #### {% linkable_title Style 2 %}
 
-If you prefer to place your entries somewhere in the `configuration.yaml` file then you just choose this style. You need to append numbers (see example below) or strings (like `media_player livingroom:` or `media_player kitchen:`) to differentiate the entries. The appended number or string must be unique.
+List each device separately.  
```
We could add this line to the title: `Style 2: List each device separately`

lindsaymarkward commented:
New commit added that addresses these issues. @balloob I didn't realised I'd changed it so it didn't show the comma separated list. That's back now. I only see two ways though. What's the third?
I'm not sure if I need to squash this new commit into the last one. Do I?

Landrash commented on source/_components/group.markdown:
```diff
@@ -42,24 +43,32 @@ group:
 </p>
 
-If all entities are switches or lights they can be controlled as one with a switch at the top of the card. Grouped states should share the same type of states (ON/OFF or HOME/NOT_HOME).
+If all entities in a group are switches or lights then Home Assistant adds a switch at the top of the card that turns them all on/off at once.
```
I have groups with a mix of devices and Home Assistant still adds a switch on top of the card. 
Groups with devices that have an **Assumed State** will have this inherited from these devices and will show the corresponding switch for the group.
